File: EJERARREGLOS.java
import java.util.Scanner;
public class EJERARREGLOS {
    static String[] meses = {
        "Enero", "Febrero", "Marzo", "Abril", "Mayo", "Junio",
        "Julio", "Agosto", "Septiembre", "Octubre", "Noviembre", "Diciembre"
    };
    static String[] departamentos = {"Ropa", "Deportes", "Juguetería"};
    static double[][] ventas = new double[12][3];

    static Scanner sc = new Scanner(System.in);
    static Integer parseMes(String input) {
        input = input.trim().toLowerCase();

        if (input.matches("\\d+")) {
            int idx = Integer.parseInt(input) - 1;
            if (idx >= 0 && idx < meses.length) return idx;
            return null;
        }
       
        for (int i = 0; i < meses.length; i++) {
            if (meses[i].toLowerCase().equals(input)) return i;
        }
    
        if (input.length() >= 2) {
            String pref = input.substring(0, Math.min(3, input.length()));
            for (int i = 0; i < meses.length; i++) {
                if (meses[i].toLowerCase().startsWith(pref)) return i;
            }
        }
        return null;
    }
    static Integer parseDepto(String input) {
        input = input.trim().toLowerCase();
      
        if (input.matches("\\d+")) {
            int idx = Integer.parseInt(input) - 1;
            if (idx >= 0 && idx < departamentos.length) return idx;
            return null;
        }
       
        for (int j = 0; j < departamentos.length; j++) {
            if (departamentos[j].toLowerCase().equals(input)) return j;
        }
      
        if (input.length() >= 2) {
            String pref = input.substring(0, Math.min(3, input.length()));
            for (int j = 0; j < departamentos.length; j++) {
                if (departamentos[j].toLowerCase().startsWith(pref)) return j;
            }
        }
        return null;
    }
    static void insertarVenta() {
        System.out.println("\n-> Insertar venta");
        System.out.print("Mes (número 1-12 o nombre): ");
        Integer i = parseMes(sc.nextLine());
        if (i == null) {
            System.out.println(" Mes no válido.");
            return;
        }
        System.out.println("Departamentos: 1-Ropa  2-Deportes  3-Juguetería");
        System.out.print("Departamento: ");
        Integer j = parseDepto(sc.nextLine());
        if (j == null) {
            System.out.println(" Departamento no válido.");
            return;
        }
        System.out.print("Monto de la venta: ");
        if (!sc.hasNextDouble()) {
            System.out.println(" Monto inválido.");
            sc.nextLine();
            return;
        }
        double monto = sc.nextDouble(); sc.nextLine();
        ventas[i][j] = monto;
        System.out.println(" Guardado: " + meses[i] + " - " + departamentos[j] + " = " + monto);
    }

    static void buscarVenta() {
        System.out.println("\n-> Buscar venta");
        System.out.print("Mes: ");
        Integer i = parseMes(sc.nextLine());
        if (i == null) {
            System.out.println(" Mes no válido.");
            return;
        }
        System.out.print("Departamento: ");
        Integer j = parseDepto(sc.nextLine());
        if (j == null) {
            System.out.println(" Departamento no válido.");
            return;
        }
        System.out.println(" Resultado: " + meses[i] + " - " + departamentos[j] + " = " + ventas[i][j]);
    }

    static void eliminarVenta() {
        System.out.println("\n-> Eliminar venta");
        System.out.print("Mes: ");
        Integer i = parseMes(sc.nextLine());
        if (i == null) {
            System.out.println(" Mes no válido.");
            return;
        }
        System.out.print("Departamento: ");
        Integer j = parseDepto(sc.nextLine());
        if (j == null) {
            System.out.println(" Departamento no válido.");
            return;
        }
        ventas[i][j] = 0;
        System.out.println(" Eliminado: " + meses[i] + " - " + departamentos[j]);
    }

    static void mostrarVentas() {
        System.out.println("\n Ventas actuales:");
        System.out.printf("%-12s%-12s%-12s%-12s%n", "Mes", "Ropa", "Deportes", "Juguetería");
        for (int i = 0; i < meses.length; i++) {
            System.out.printf("%-12s%-12.2f%-12.2f%-12.2f%n",
                    meses[i], ventas[i][0], ventas[i][1], ventas[i][2]);
        }
    }

    public static void main(String[] args) {
        while (true) {
            System.out.println("\n===== MENÚ =====");
            System.out.println("1) Insertar venta");
            System.out.println("2) Buscar venta");
            System.out.println("3) Eliminar venta");
            System.out.println("4) Mostrar todas las ventas");
            System.out.println("5) Salir");
            System.out.print("Opción: ");
            String opc = sc.nextLine().trim();
            switch (opc) {
                case "1": insertarVenta(); break;
                case "2": buscarVenta(); break;
                case "3": eliminarVenta(); break;
                case "4": mostrarVentas(); break;
                case "5": System.out.println("Saliendo"); return;
                default: System.out.println(" Opción inválida.");
            }
        }
    }
}

File: EJER - ARREGLOS.py

meses = [
    "Enero", "Febrero", "Marzo", "Abril", "Mayo", "Junio",
    "Julio", "Agosto", "Septiembre", "Octubre", "Noviembre", "Diciembre"
]
departamentos = ["Ropa", "Deportes", "Juguetería"]
ventas = [[0 for _ in departamentos] for _ in meses]

def parse_mes(inp):
    s = inp.strip().lower()
   
    if s.isdigit():
        i = int(s) - 1
        if 0 <= i < len(meses):
            return i
        return None
    
    for i, m in enumerate(meses):
        if m.lower() == s:
            return i

    if len(s) >= 2:
        pref = s[:3]
        for i, m in enumerate(meses):
            if m.lower().startswith(pref):
                return i
    return None
def parse_depto(inp):
    s = inp.strip().lower()
    
    if s.isdigit():
        j = int(s) - 1
        if 0 <= j < len(departamentos):
            return j
        return None
   
    for j, d in enumerate(departamentos):
        if d.lower() == s:
            return j
    
    if len(s) >= 2:
        pref = s[:3]
        for j, d in enumerate(departamentos):
            if d.lower().startswith(pref):
                return j
    return None
def insertar_venta():
    print("\n-> Insertar venta")
    print("Escribe el mes (número 1-12 o nombre/abreviatura). Eje: 2 o feb")
    mes_in = input("Mes: ")
    i = parse_mes(mes_in)
    if i is None:
        print("Mes no válido. Intenta de nuevo.")
        return
    print("Departamentos: 1-Ropa  2-Deportes  3-Juguetería")
    depto_in = input("Departamento (número o nombre): ")
    j = parse_depto(depto_in)
    if j is None:
        print("Departamento no válido. Intenta de nuevo.")
        return
    try:
        monto = float(input("Ingresa el monto de la venta (eje. 1500): "))
    except ValueError:
        print("Monto no válido. Debe ser un número.")
        return
    ventas[i][j] = monto  
    print(f" Guardado: {meses[i]} - {departamentos[j]} = {monto}")
def buscar_venta():
    print("\n Buscar venta")
    mes_in = input("Mes (número o nombre): ")
    i = parse_mes(mes_in)
    if i is None:
        print("Mes no válido.")
        return
    depto_in = input("Departamento (número o nombre): ")
    j = parse_depto(depto_in)
    if j is None:
        print("Departamento no válido.")
        return
    print(f" Resultado: {meses[i]} - {departamentos[j]} = {ventas[i][j]}")
def eliminar_venta():
    print("\n-> Eliminar venta")
    mes_in = input("Mes (número): ")
    i = parse_mes(mes_in)
    if i is None:
        print("Mes no válido.")
        return
    depto_in = input("Departamento (número): ")
    j = parse_depto(depto_in)
    if j is None:
        print("Departamento no válido.")
        return
    ventas[i][j] = 0
    print(f" Eliminado: {meses[i]} - {departamentos[j]}")
def mostrar_ventas():
    print("\n Ventas actuales:")
    header = f"{'Mes':<12}" + "".join(f"{d:<12}" for d in departamentos)
    print(header)
    for i, mes in enumerate(meses):
        fila = f"{mes:<12}" + "".join(f"{ventas[i][j]:<12}" for j in range(len(departamentos)))
        print(fila)
def menu():
    while True:
        print("\n MENÚ ")
        print("1) Insertar venta")
        print("2) Buscar venta")
        print("3) Eliminar venta")
        print("4) Mostrar todas las ventas")
        print("5) Salir")
        opc = input("Elige una opción (1-5): ").strip()
        if opc == "1":
            insertar_venta()
        elif opc == "2":
            buscar_venta()
        elif opc == "3":
            eliminar_venta()
        elif opc == "4":
            mostrar_ventas()
        elif opc == "5":
            print("saliendo.")
            break
        else:
            print("Opción inválida. Escribe 1, 2, 3, 4 o 5.")

if __name__ == "__main__":
    menu()
